sn_registers/src/register.rs

```diff
@@ -234,17 +223,15 @@ impl Register {
     }
 
     /// Merge another Register into this one.
-    pub fn merge(&mut self, other: Self) {
-        self.crdt.merge(other.crdt);
+    pub fn merge(&mut self, other: &Self) -> Result<()> {
```

Comment:
we could replace the function name with an overview of what the comment above is saying, removing the need for the comment and making the function name more descriptive

Reply:
i.e merge_register